arangod/RocksDBEngine/RocksDBVPackIndex.cpp

```diff
@@ -3205,4 +3206,117 @@ Index::StreamSupportResult RocksDBVPackIndex::supportsStreamInterface(
                                       streamOpts);
 }
 
+bool RocksDBVPackIndex::checkSupportScanDistinct(
+    IndexDistinctScanOptions const& options,
+    std::vector<std::vector<basics::AttributeName>> const& fields) noexcept {
+  auto distinctFields = options.distinctFields;
+  if (not options.sorted) {
+    // if no sorted output is required, we can change the order of fields
+    std::sort(distinctFields.begin(), distinctFields.end());
+  }
+
+  // we only support distinct scans for a prefix of the fields
+  for (size_t k = 0; k < distinctFields.size(); k++) {
+    if (k >= fields.size()) {
+      return false;
+    }
+    if (k != distinctFields[k]) {
+      return false;
+    }
+  }
+
+  return true;
+}
```

Comment:
I'm also still wandering why `_fields` is never used in this function, see [Tobias comment](https://github.com/arangodb/arangodb/pull/21541#discussion_r1907363213).